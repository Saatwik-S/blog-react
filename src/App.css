* {
  margin: 0;
  padding: 0;
  font-family:Cambria, Cochin, Georgia, Times, 'Times New Roman', serif ;
}

.paddings {
  padding: 0 10% 0 10%;
}

.body {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

header {
  font-size: x-large;

  color: white;
  background-color: black;
  display: flex;
  height: 10vh;
  justify-content: space-between;
  align-items: center;
}
nav {
  width: 100%;
  height: inherit;
  display: flex;
  align-items: flex-end;
  justify-content: flex-end;
}

nav > a {
  color: white;
  padding: 0 15px 10px 15px;
  text-decoration: none;
}

/* Main body */
.main-body {
  display: flex;
  flex-wrap:  wrap;
  margin: 20px;
  justify-content: space-between;
}

/* Header Title */
.title {
  padding: 15px;
  width: 30vw;
  height: 70%;
  display: flex; 
  align-items: flex-start;

}

/* Footer */
footer {
  display: flex;
  background-color: black;
  height: 50px;
  color: white;
  align-items: center;
}
footer > p {
  font-size: small;
  padding: 10px 10px 10px 10px;
}

/* Information Box */
.display-box {
  height: 500px;
  border: 1px ridge black;
  width: 30%;
  margin-top: 30px;
  /* flex-basis: 25%; */
}

@media screen and (max-width: 1200px) {
  .display-box {
      flex-direction: column;
      width:45%;
      align-items: center;
     
  }
  .box-title {
      font-size: 23;
  }
  
}

@media screen and (max-width: 720px) {
   .display-box {
      flex-direction: column;
      width:80%;

          
  }

  .main-body{
      display: flex;
      justify-content: center;
  }

  .title {
      font-size: medium;
  } 
}
  

  


.box-data {
  height:50%;
  padding: 0 5px;
}

.display-box > img {
  width: 100%;
  height: 50%;
  object-fit: fill;
}


.box-date {
  height: 10%;
  display: flex;
  justify-content: space-between;
  font-size: 15px;
  color: grey;
}

.box-title {

  height: 30%;
  word-wrap: break-word;
  text-overflow: ellipsis;
  font-size: 25px;
}

.box-desc {
  height: 40%;
  max-height: 40%;
  color: grey;
  font-size: larger;
  text-overflow: ellipsis;
  overflow-y: scroll;
}
.box-divider {
   height: 1%;
  background-color: grey;
}

.box-likes {
  height: 18%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  
}

.claps > p {
  color: grey;
  margin-left: 2px ;
}
.claps {
  display: flex;

}
/* ICONS */

.icon-heart-red {
  background: url("./assets/icons/heart-red.svg");
  height: 20px;
  width: 20px;
  display: block;
}

.icon-heart-black {
  background: url("./assets/icons/heart-black.svg");
  height: 20px;
  width: 20px;
  display: block;
}
.icon-clap {
  background: url("./assets/icons/clapping.svg");
  height: 20px;
  width: 20px;
  display: block;
}
