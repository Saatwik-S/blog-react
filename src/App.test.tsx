import React from 'react'
import { render } from '@testing-library/react'
import App from './App'

describe('App', () => {
  test('Renders the App Component', () => {
    const { asFragment } = render(<App />)
    expect(asFragment()).toMatchSnapshot()
  })
})
