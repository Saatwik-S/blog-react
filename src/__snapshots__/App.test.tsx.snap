// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the App Component 1`] = `
<DocumentFragment>
  <div
    class="body"
  >
    <header
      class="paddings"
    >
      <div
        class="title"
      >
        <p>
          The Artifact
          <br />
          <em>
             Culture & Art Blog 
          </em>
        </p>
      </div>
      <nav>
        <a
          href="index.html"
        >
          Blog
        </a>
        <a
          href="about.html"
        >
          About
        </a>
        <a
          href="contact.html"
        >
          Contact
        </a>
      </nav>
    </header>
    <div
      class="main-body paddings"
    >
      <div
        class="display-box"
        data-testid="box"
      >
        <img
          src="/images/abstract.png"
        />
        <div
          class="box-data"
        >
          <div
            class="box-date"
          >
            <p>
              2nd Januray, 2018
            </p>
            <p>
              2 mins
            </p>
          </div>
          <div
            class="box-title"
          >
            <p>
              The future of abstract art and the culture ...
            </p>
          </div>
          <div
            class="box-desc"
          >
            <p>
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <div
            class="box-divider"
          />
          <div
            class="box-likes"
          >
            <div
              class="claps"
              data-testid="clap-button"
            >
              <i
                class="icon-clap"
              />
              <p>
                10
              </p>
            </div>
            <i
              class="icon-heart-black"
              data-testid="like-button"
            />
          </div>
        </div>
      </div>
      <div
        class="display-box"
        data-testid="box"
      >
        <img
          src="/images/outdoor.png"
        />
        <div
          class="box-data"
        >
          <div
            class="box-date"
          >
            <p>
              31st Januray, 2018
            </p>
            <p>
              1 min
            </p>
          </div>
          <div
            class="box-title"
          >
            <p>
              How painting outdoors can transform your 
            </p>
          </div>
          <div
            class="box-desc"
          >
            <p>
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <div
            class="box-divider"
          />
          <div
            class="box-likes"
          >
            <div
              class="claps"
              data-testid="clap-button"
            >
              <i
                class="icon-clap"
              />
              <p>
                1
              </p>
            </div>
            <i
              class="icon-heart-red"
              data-testid="like-button"
            />
          </div>
        </div>
      </div>
      <div
        class="display-box"
        data-testid="box"
      >
        <img
          src="/images/young-painters.png"
        />
        <div
          class="box-data"
        >
          <div
            class="box-date"
          >
            <p>
              20th March, 2018
            </p>
            <p>
              2 mins
            </p>
          </div>
          <div
            class="box-title"
          >
            <p>
              10 young painters you need to know
            </p>
          </div>
          <div
            class="box-desc"
          >
            <p>
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <div
            class="box-divider"
          />
          <div
            class="box-likes"
          >
            <div
              class="claps"
              data-testid="clap-button"
            >
              <i
                class="icon-clap"
              />
              <p>
                15
              </p>
            </div>
            <i
              class="icon-heart-black"
              data-testid="like-button"
            />
          </div>
        </div>
      </div>
      <div
        class="display-box"
        data-testid="box"
      >
        <img
          src="/images/perfect-art.png"
        />
        <div
          class="box-data"
        >
          <div
            class="box-date"
          >
            <p>
              22nd June, 2018
            </p>
            <p>
              5 mins
            </p>
          </div>
          <div
            class="box-title"
          >
            <p>
              Tips for creating the perfect art
            </p>
          </div>
          <div
            class="box-desc"
          >
            <p>
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <div
            class="box-divider"
          />
          <div
            class="box-likes"
          >
            <div
              class="claps"
              data-testid="clap-button"
            >
              <i
                class="icon-clap"
              />
              <p>
                11
              </p>
            </div>
            <i
              class="icon-heart-black"
              data-testid="like-button"
            />
          </div>
        </div>
      </div>
      <div
        class="display-box"
        data-testid="box"
      >
        <img
          src="/images/inspired.png"
        />
        <div
          class="box-data"
        >
          <div
            class="box-date"
          >
            <p>
              4th Feburary, 2019
            </p>
            <p>
              9 mins
            </p>
          </div>
          <div
            class="box-title"
          >
            <p>
              Stay inspired: top artists blog ...
            </p>
          </div>
          <div
            class="box-desc"
          >
            <p>
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <div
            class="box-divider"
          />
          <div
            class="box-likes"
          >
            <div
              class="claps"
              data-testid="clap-button"
            >
              <i
                class="icon-clap"
              />
              <p>
                45
              </p>
            </div>
            <i
              class="icon-heart-red"
              data-testid="like-button"
            />
          </div>
        </div>
      </div>
      <div
        class="display-box"
        data-testid="box"
      >
        <img
          src="/images/reason.png"
        />
        <div
          class="box-data"
        >
          <div
            class="box-date"
          >
            <p>
              10th Januray, 2019
            </p>
            <p>
              2 mins
            </p>
          </div>
          <div
            class="box-title"
          >
            <p>
              5 reasons why everyone should try an...
            </p>
          </div>
          <div
            class="box-desc"
          >
            <p>
              Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
            </p>
          </div>
          <div
            class="box-divider"
          />
          <div
            class="box-likes"
          >
            <div
              class="claps"
              data-testid="clap-button"
            >
              <i
                class="icon-clap"
              />
              <p>
                11
              </p>
            </div>
            <i
              class="icon-heart-black"
              data-testid="like-button"
            />
          </div>
        </div>
      </div>
    </div>
    <footer
      class="paddings"
    >
      <p>
        Â©artifact.com 2019
      </p>
    </footer>
  </div>
</DocumentFragment>
`;
